esx_service/utils/auth_api.py

```diff
@@ -150,32 +197,30 @@ def generate_privileges(datastore, allow_create, volume_maxsize, volume_totalsiz
     if allow_create is True:
         set_privileges(allow_create, privileges, 1)
 
-    if volume_maxsize:
-        size_in_MB = convert.convert_to_MB(volume_maxsize)
-        privileges[auth_data_const.COL_MAX_VOLUME_SIZE] = size_in_MB
+    if volume_maxsize_in_MB:
+        privileges[auth_data_const.COL_MAX_VOLUME_SIZE] = volume_maxsize_in_MB
 
-    if volume_totalsize:
-        size_in_MB = convert.convert_to_MB(volume_totalsize)
-        privileges[auth_data_const.COL_USAGE_QUOTA] = size_in_MB
+    if volume_totalsize_in_MB:
+        privileges[auth_data_const.COL_USAGE_QUOTA] = volume_totalsize_in_MB
     
-    logging.debug("generate_privileges: privileges=%s", privileges)
+    logging.debug("generate_privileges: privileges=%s", privileges) 
     return privileges
 
-def modify_privileges(privileges, allow_create, volume_maxsize, volume_totalsize):
+def modify_privileges(privileges, allow_create, volume_maxsize_in_MB, volume_totalsize_in_MB):
     """ Modify privileges based on input params """
+    logging.debug("modify_privileges: allow_create=%s, volume_maxsize_in_MB=%s, volume_totalsize_in_MB=%s", 
+                  allow_create, volume_maxsize_in_MB, volume_totalsize_in_MB)
     if allow_create is not None:
-        if allow_create is True:
-            set_privileges(allow_create, privileges, 1)
+        if allow_create == "True":
+            privileges = set_privileges(allow_create, privileges, 1)
         else:
-            set_privileges(allow_create, privileges, 0)
+            privileges = set_privileges(allow_create, privileges, 0)
```

Comment:
This function is used to modify an existing privilege. "allow_create" is passed by user to specify if he/she wants to modify this field in the passed-in "privileges". "allow_create" is None just means user does not want to change the "allow_create" filed in the passed-in "privileges".